dask_glm/logistic.py

```diff
@@ -0,0 +1,380 @@
+from __future__ import absolute_import, division, print_function
+
+from dask import delayed, persist, compute
+import numpy as np
+import dask.array as da
+from numba import jit
+from scipy.optimize import fmin_l_bfgs_b
+
+from dask_glm.utils import dot, exp, log1p
+
+
+def bfgs(X, y, max_iter=50, tol=1e-14):
+    '''Simple implementation of BFGS.'''
+
+    n, p = X.shape
+
+    y_local = da.compute(y)[0]
+
+    recalcRate = 10
+    stepSize = 1.0
+    armijoMult = 1e-4
+    backtrackMult = 0.1
+
+    beta = np.zeros(p)
+    Hk = np.eye(p)
+    sk = None
+    for k in range(max_iter):
+
+        if k % recalcRate == 0:
+            Xbeta = X.dot(beta)
+            eXbeta = exp(Xbeta)
+            func = log1p(eXbeta).sum() - y.dot(Xbeta)
+
+        e1 = eXbeta + 1.0
+        gradient = X.T.dot(
+            eXbeta / e1 - y)  # implicit numpy -> dask conversion
+
+        if k:
+            yk = yk + gradient  # TODO: gradient is dasky and yk is numpy-y
+            rhok = 1 / yk.dot(sk)
+            adj = np.eye(p) - rhok * dot(sk, yk.T)
+            Hk = dot(adj, dot(Hk, adj.T)) + rhok * dot(sk, sk.T)
+
+        step = dot(Hk, gradient)
+        steplen = dot(step, gradient)
+        Xstep = dot(X, step)
+
+        Xbeta, gradient, func, steplen, step, Xstep = da.compute(
+            Xbeta, gradient, func, steplen, step, Xstep)
```

Comment:
We want to avoid ever calling `compute` on a large result (this causes cluster-to-client communication).  Instead we're starting to use persist